### ISSUE TITLE
Invalid date format

### ISSUE THREAD

**Issue by mark-prins** [openmsupply/open-msupply#2107]
## What went wrong? 😲
 
Noticed when testing #2066 
When you add a new encounter the date format is different to the rest of the site which is using `DD/MM/YYYY` - for encounters, you can enter the following as a valid date:

<img width="381" alt="Screenshot 2023-08-18 at 9 22 00 AM" src="https://github.com/openmsupply/open-msupply/assets/9192912/b4949b4c-97fc-44ed-8bdd-89fb7c7db8df">

I would also like to have the method implemented which was raised in [this comment](https://github.com/openmsupply/open-msupply/pull/2066#discussion_r1292858650) - to expand out this usage:

```
t(`error.date_${validationError}` as LocaleKey, {
  defaultValue: validationError,
})
```

into the exhaustive and more typesafe version:

```
const formattedDateError = (t: TypedTFunction<LocaleKey>, error: string) => {
  if (!error) return null;

  switch (error) {
    case 'date_invalidDate':
      return t('error.date_invalidDate');
    case 'date_minDate':
      return t('error.date_minDate');
    case 'date_maxDate':
      return t('error.date_maxDate');
    case 'date_disablePast':
      return t('error.date_disablePast');
    case 'date_disableFuture':
      return t('error.date_disableFuture');
    default:
      return error;
  }
};
```

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

 - Version: 
 - Platform:
   - [ ] android (tablet)
   - [ ] browser (extra points if you tell us which one)
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

**Comment by roxy-dao** [comment]
Also getting RangeError for stocktakes if the stock line has an expiry date

**Comment by CarlosNZ** [comment]
Closed by #2110
